services/src/main/scala/cromwell/services/cost/CostCatalogService.scala

```diff
@@ -0,0 +1,110 @@
+package cromwell.services.cost
+
+import akka.actor.{Actor, ActorRef}
+import com.google.cloud.billing.v1._
+import com.typesafe.config.Config
+import com.typesafe.scalalogging.LazyLogging
+import common.util.StringUtil.EnhancedToStringable
+import cromwell.services.ServiceRegistryActor.ServiceRegistryMessage
+import cromwell.util.GracefulShutdownHelper.ShutdownCommand
+
+import java.io.{FileOutputStream, ObjectOutputStream}
+import java.time.{Duration, LocalDateTime}
+import scala.jdk.CollectionConverters.IterableHasAsScala
+import java.time.temporal.ChronoUnit.HOURS
+case class CostCatalogMessage() extends ServiceRegistryMessage {
+  override val serviceName = "CostCatalogService"
+}
+
+case class ExpiringCostCatalog(catalog: Map[CostCatalogKey, CostCatalogValue], fetchTime: LocalDateTime)
+
+/**
+ * This actor handles the fetching and caching of the public Google Cost Catalog.
+ * This enables us to look up the SKUs necessary for cost calculations.
+ */
+class CostCatalogService(serviceConfig: Config,
+                         globalConfig: Config,
+                         serviceRegistry: ActorRef,
+                         mockTestData: Option[List[Sku]] = Option.empty,
+                         maxCatalogLifetime: Duration = Duration.of(24, HOURS)
+) extends Actor
+    with LazyLogging {
+
+  // Cached catalog. Refreshed lazily when older than maxCatalogLifetime.
+  private var costCatalog: ExpiringCostCatalog =
+    ExpiringCostCatalog(CostCatalogService.processCostCatalog(fetchNewCatalog), java.time.LocalDateTime.now())
+
+  /**
+   * Returns the SKU for a given key, if it exists
+   */
+  def getSku(key: CostCatalogKey): Option[CostCatalogValue] = getOrFetchCachedCatalog().get(key)
+
+  def serviceRegistryActor: ActorRef = serviceRegistry
+  override def receive: Receive = {
+    case ShutdownCommand => context stop self
```

Comment:
Looks like the Google client has a `shutdownNow()` method, may be useful to put in here to immediately cancel any requests.